server/desktop/src/main/java/dev/slimevr/desktop/tracking/trackers/hid/TrackersHID.kt

```diff
@@ -182,20 +184,17 @@ class TrackersHID(name: String, private val trackersConsumer: Consumer<Tracker>)
 						val idCombination = dataReceived[i + 1].toInt()
 						val rssi = -dataReceived[i + 2].toInt()
 						val battery = dataReceived[i + 3].toInt()
-						val battery_mV = dataReceived[i + 5].toInt() and 255 shl 8 or (dataReceived[i + 4].toInt() and 255)
-						val q = floatArrayOf(0f, 0f, 0f, 0f)
-						val a = floatArrayOf(0f, 0f, 0f)
-						for (j in 0..3) { // quat received as fixed 14
-							var buf =
-								dataReceived[i + 6 + j * 2 + 1].toInt() and 255 shl 8 or (dataReceived[i + 6 + j * 2].toInt() and 255)
-							buf -= 32768 // uint to int
-							q[j] = buf / (1 shl 14).toFloat() // fixed 14 to float
+						// ushort big endian
+						val battery_mV = dataReceived[i + 5].toUByte().toInt() shl 8 or dataReceived[i + 4].toUByte().toInt()
+						// Q15: 1 is represented as 0x7FFF, -1 as 0x8000
```

Comment:
Originally used Q14 to not worry about representing 1 but it is probably not important. You cannot really represent exactly 1 with Q15 format, but the nearest value is good enough

Reply:
The quaternion `(0, 0, 1, 0)` will be encoded to `(0, 0, 7FFF, 0)`. If one decode it and normalize it will be `(0, 0, 1, 0)` up to a single ULP.

Q15 is twice (one bit) more precise for all values except for four values like `(0, 0, 1, 0)`.

Plus, Q15 is a standard for `[-1, 1]` values encoding in 16-bits.

And again, unification with #1001.